src/sdk/pynni/nni/pbt_tuner/pbt_tuner.py

```diff
@@ -0,0 +1,251 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""
+pbt_tuner.py
+"""
+
+import copy
+import logging
+import os
+import numpy as np
+
+import nni
+from nni.tuner import Tuner
+from nni.utils import OptimizeMode, extract_scalar_reward, split_index, json2parameter, json2space
+
+
+logger = logging.getLogger('pbt_tuner_AutoML')
+
+
+def exploit_and_explore(bot_trial_info, top_trial_info, factors, epoch):
+    """
+    Replace checkpoint of bot_trial with top, and perturb hyperparameters
+
+    """
+    bot_checkpoint_dir = bot_trial_info.checkpoint_dir
+    top_hyper_parameters = top_trial_info.hyper_parameters
+    hyper_parameters = copy.deepcopy(top_hyper_parameters)
+    for key in hyper_parameters.keys():
+        if key == 'load_checkpoint_dir':
+            hyper_parameters[key] = hyper_parameters['save_checkpoint_dir']
+        elif key == 'save_checkpoint_dir':
+            hyper_parameters[key] = os.path.join(bot_checkpoint_dir, str(epoch))
+        elif isinstance(hyper_parameters[key], float):
```

Comment:
why not perturb other types of hyper-parameters such as int, string?

Reply:
In the paper, this way of exploration was introduced, but it is not applicable to other types of data. I am not sure how to perturb other types of data. Maybe I can add them in the future.

Reply:
good point @leckie-chn . @RayMeng8 if you want to support other types in future, please make it clear what types of search space does PBT support. https://github.com/microsoft/nni/blob/master/docs/en_US/Tutorial/SearchSpaceSpec.md#search-space-types-supported-by-each-tuner